src/components/common/dropdown/MenuDropdown.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+} from "@/components/ui/dropdown-menu";
+import { cn } from "@/lib/utils";
+
+interface Option {
+  label: React.ReactNode;
+  value: string;
+  ariaLabel?: string;
+}
+
+/**
+ * MenuDropdown
+ * - 수정/삭제 같은 액션을 위한 메뉴 드롭다운 컴포넌트
+ * - trigger prop으로 원하는 요소를 넘겨서 사용
+ * - 옵션 label에 ReactNode 허용 (이미지, 아이콘 등)
+ */
+interface MenuDropdownProps {
+  options: Option[]; // 표시할 메뉴 항목들
+  onSelect: (value: string) => void; // 항목 클릭 시 실행할 콜백
+  trigger: React.ReactElement;
+}
```

Comment:
`onSelect`콜백의 파라미터 value는 아래 사용하는 곳을 보니 Option['value'] 타입이군요. 제네릭을 쓰면 더 타입안전한 컴포넌트가 될 수 있겠네요! TS역량을 어필할 수도 있고요 (제네릭은  ts코드를 - 특히 라이브러리나 기존의 코드 - 이해하는데 필수입니다)

```ts
onClick={() => onSelect(option.value)}
```

```suggestion

interface Option<T> {
  label: React.ReactNode;
  value: T;
  ariaLabel?: string;
}

interface MenuDropdownProps<T> {
  options: Option<T>[]; // 표시할 메뉴 항목들
  onSelect: (value: T) => void; // 항목 클릭 시 실행할 콜백
  trigger: React.ReactElement;
}

export default function MenuDropdown<T extends string>({
  options,
  onSelect,
  trigger,
}: MenuDropdownProps<T>) {
```